protocol/shannon/protocol.go

```diff
@@ -296,7 +325,21 @@ func (p *Protocol) getSessionsUniqueEndpoints(
 		serviceID, len(activeSessions),
 	)
 
+	// Get fallback configuration for the service ID.
+	// Fallback configuration will be used in the following scenarios:
+	// 	 1. The service is configured to send all traffic to fallback endpoints.
+	// 	 2. All service endpoints are sanctioned and the service is configured with at least one fallback endpoint.
+	fallbackEndpoints, sendAllTrafficToFallback := p.getServiceFallbackConfig(serviceID)
+
+	// If the service is configured to send all traffic to fallback endpoints,
+	// return only the fallback endpoints and skip the rest of the logic.
+	if sendAllTrafficToFallback && len(fallbackEndpoints) > 0 {
```

Comment:
IMO this SHOULD NOT be part of `getSessionUniqueEndpoints`.

We're hiding/embedding logic.

I'd prefer this return an empty list/array and the caller will then retrieve fallbackEndpoints.

cc @adshmh for another POV.

Reply:
You can create a `getUniqueEndpoints` that wraps this logic.

Reply:
This is fine, and I agree it makes it cleaner. I've refactored to separate session endpoint logic.

Reply:
Agreed that making the logic more explicit and visible is a good change here.